tests/mock_network/churn.rs

```diff
@@ -571,30 +577,34 @@ fn messages_during_churn() {
             max_iterations,
             current_sections(&nodes).format(", ")
         );
-        let new_indices = random_churn(
+        let (added_indices, dropped_names) = random_churn(
             &mut rng,
             &network,
             &mut nodes,
             churn_probability,
             min_section_num,
             max_section_num,
         );
+
+        if !dropped_names.is_empty() {
```

Comment:
I would remove the if and do it always, makes it easier to read logs, and simplify test code a tiny bit.
Same for Added nodes.